src/RpcClient/Utility.cs

```diff
@@ -142,7 +142,7 @@ public static Header HeaderFromJson(JObject json, ProtocolSettings protocolSetti
                 PrevHash = UInt256.Parse(json["previousblockhash"].AsString()),
                 MerkleRoot = UInt256.Parse(json["merkleroot"].AsString()),
                 Timestamp = (ulong)json["time"].AsNumber(),
-                Nonce = (ulong)json["nonce"].AsNumber(),
+                Nonce = UInt64.Parse(json["nonce"].AsString()),
```

Comment:
I think you should use`UInt64.Parse(json["nonce"].AsString(), NumberStyles.AllowHexSpecifier)` to convert a hex string to uint64.